File: app/src/main/java/com/fanwe/json/MainActivity.java
package com.fanwe.json;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.fanwe.lib.utils.json.FJson;

public class MainActivity extends AppCompatActivity
{
    public static final String TAG = MainActivity.class.getSimpleName();

    private final String json = "{\"valueInt\":\"1\", \"valueLong\":\"2\", \"valueFloat\":\"3.3\", \"valueDouble\":\"4.4\"}";

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TestModel model = FJson.jsonToObject(json, TestModel.class);
        Log.i(TAG, model.toString());
    }
}

File: app/src/main/java/com/sd/demo/json/MainActivity.kt
package com.sd.demo.json

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.sd.demo.json.databinding.ActivityMainBinding
import com.sd.demo.json.model.PageRequestParams
import com.sd.demo.json.model.UserModel
import com.sd.lib.json.FJson

class MainActivity : AppCompatActivity() {
    private val _binding by lazy { ActivityMainBinding.inflate(layoutInflater) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(_binding.root)

        _binding.btnEncode.setOnClickListener {
            encodeRequestParams()
        }

        _binding.btnDecode.setOnClickListener {
            decodeUser()
        }
    }
}

private fun decodeUser() {
    val json = """
            {"id":"1","name":null,"age":null}
        """.trimIndent()
    val model = FJson.fromJson(json, UserModel::class.java)
    logMsg {
        "decodeDataUser:${model}"
    }
}

private fun encodeRequestParams() {
    val model = PageRequestParams().apply {
        this.put("key", "value")
        this.setPage(1)
    }
    val json = FJson.toJson(model, PageRequestParams::class.java)
    logMsg {
        "encodeRequestParams:$json"
    }
}

inline fun logMsg(block: () -> String) {
    Log.i("json-demo", block())
}